File: java/insertionSort.java
import java.util.Scanner;
class insertionSort{
	static void sort(int arr[])  // method to sort the array
	{
		int n = arr.length;
		for (int i = 1; i < n; ++i) {
			int key = arr[i];
			int j = i - 1;
			while (j >= 0 && arr[j] > key) {
				arr[j + 1] = arr[j];
				j = j - 1;
			}
			arr[j + 1] = key;
		}
	}

	
	static void printArray(int arr[]) // method to print the array
	{
		int n = arr.length;
		for (int i = 0; i < n; ++i)
			System.out.print(arr[i] + " ");

		System.out.println();
	}

	
	public static void main(String args[])
	{
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter array size: ");
		int n=sc.nextInt();
		int arr[]=new int[n];
		System.out.println("Enter array elements: ");
		for(int i=0;i<n;i++){
			arr[i]=sc.nextInt();
		}
		sort(arr);
		System.out.println("Sorted array: ");
		printArray(arr);
	}
}

/*
Example Output:
Enter array size: 
5
Enter array elements: 
123 45 76 87 94
Sorted array: 
45 76 87 94 123 

*/

File: python/Data Structures/Sorting/insertion_sort.py
# Function to do insertion sort
def insertionSort(arr):
	for i in range(1, len(arr)):
		key = arr[i]
		j = i-1
		while j >=0 and key < arr[j] :
				arr[j+1] = arr[j]
				j -= 1
		arr[j+1] = key
		
n=int(input("Enter array size: "))
print("Enter the array elements: ")
arr=list(map(int,input().split()))
insertionSort(arr)
print ("Sorted array is:")
for i in range(len(arr)):
	print(arr[i],end=" ")

'''
Example Output:
Enter array size: 
5
Enter the array elements: 
123 45 76 87 94
Sorted array is: 
45 76 87 94 123 
'''